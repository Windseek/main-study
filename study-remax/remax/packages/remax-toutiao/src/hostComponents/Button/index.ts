import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface ButtonProps extends BaseProps {
  size?: 'default' | 'mini';
  type?: 'primary' | 'default';
  disabled?: boolean;
  loading?: boolean;
  hoverClassName?: string;
  hoverStartTime?: number;
  hoverStayTime?: number;
  hoverStopPropagation?: boolean;
  formType?: 'submit' | 'reset';
  openType?: 'share' | 'getPhoneNumber';
  onClick?: (e: any) => void;
  onGetPhoneNumber?: (e: any) => void;
}

export const Button: React.ComponentType<ButtonProps> = createHostComponent<ButtonProps>('button');

Button.defaultProps = {
  size: 'default',
  type: 'default',
  disabled: false,
  loading: false,
  hoverClassName: 'button-hover',
  hoverStartTime: 20,
  hoverStayTime: 70,
  hoverStopPropagation: false,
};
