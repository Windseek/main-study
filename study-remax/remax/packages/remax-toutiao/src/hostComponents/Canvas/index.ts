import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface CanvasProps extends BaseProps {
  canvasId: string;
}

export const Canvas: React.ComponentType<CanvasProps> = createHostComponent<CanvasProps>('canvas');
