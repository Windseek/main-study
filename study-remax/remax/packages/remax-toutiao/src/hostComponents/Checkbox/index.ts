import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface CheckboxProps extends BaseProps {
  name?: string;
  value?: string;
  disabled?: boolean;
  checked?: boolean;
  color?: string;
}

export const Checkbox: React.ComponentType<CheckboxProps> = createHostComponent<CheckboxProps>('checkbox');

Checkbox.defaultProps = {
  disabled: false,
  checked: false,
};
