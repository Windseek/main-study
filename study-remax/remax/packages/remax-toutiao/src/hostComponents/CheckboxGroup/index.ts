import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface CheckboxGroupProps extends BaseProps {
  name?: string;
  onChange?: (e: any) => void;
}

export const CheckboxGroup: React.ComponentType<CheckboxGroupProps> = createHostComponent<CheckboxGroupProps>(
  'checkbox-group'
);
