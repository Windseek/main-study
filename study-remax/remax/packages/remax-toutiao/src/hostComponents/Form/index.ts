import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface FormProps extends BaseProps {
  onSubmit?: (e: any) => void;
  onReset?: (e: any) => void;
  reportSubmit?: boolean;
  catchSubmit?: (e: any) => void;
  catchReset?: (e: any) => void;
}

export const Form: React.ComponentType<FormProps> = createHostComponent<FormProps>('form');

Form.defaultProps = {
  reportSubmit: false,
};
