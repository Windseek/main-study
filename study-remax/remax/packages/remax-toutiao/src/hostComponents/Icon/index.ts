import * as React from 'react';
import { createHostComponent } from '@remax/runtime';

import { BaseProps } from '../../types/component';

export interface IconProps extends BaseProps {
  type:
    | 'info'
    | 'warn'
    | 'waiting'
    | 'cancel'
    | 'download'
    | 'search'
    | 'clear'
    | 'success'
    | 'success_no_circle'
    | 'loading';
  size?: number;
  color?: string;
}

export const Icon: React.ComponentType<IconProps> = createHostComponent<IconProps>('icon');

Icon.defaultProps = {
  size: 24,
};
