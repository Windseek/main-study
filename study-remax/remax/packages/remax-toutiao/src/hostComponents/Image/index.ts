import * as React from 'react';

import { BaseProps } from '../../types/component';
import { createHostComponent } from '@remax/runtime';

export interface ImageProps extends BaseProps {
  src?: string;
  mode?:
    | 'scaleToFill'
    | 'aspectFit'
    | 'aspectFill'
    | 'widthFix'
    | 'top'
    | 'bottom'
    | 'center'
    | 'left'
    | 'right'
    | 'top left'
    | 'top right'
    | 'bottom left'
    | 'bottom right';
  style?: React.CSSProperties;
  lazyLoad?: boolean;
  onLoad?: (e: any) => void;
  onError?: (e: any) => void;
  onTap?: (e: any) => void;
  onClick?: (e: any) => void;
  onTouchStart?: (e: any) => void;
  onTouchMove?: (e: any) => void;
  onTouchEnd?: (e: any) => void;
  onTouchCancel?: (e: any) => void;
}

export const Image: React.ComponentType<ImageProps> = createHostComponent<ImageProps>('image');

Image.defaultProps = {
  mode: 'scaleToFill',
  lazyLoad: false,
};
