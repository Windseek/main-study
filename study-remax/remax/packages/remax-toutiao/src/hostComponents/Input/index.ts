import * as React from 'react';

import { BaseProps } from '../../types/component';
import { createHostComponent } from '@remax/runtime';

export interface InputProps extends BaseProps {
  name?: string;
  focus?: boolean;
  maxlength?: number;
  value?: string;
  password?: boolean;
  type?: 'text' | 'number' | 'digit';
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  adjustPosition?: boolean;
  placeholder?: string;
  placeholderStyle?: React.CSSProperties;
  placeholderClassName?: string;
  disabled?: boolean;
  cursorSpacing?: number;
  onInput?: (e: any) => void | string;
  onClick?: (e: any) => void;
  onFocus?: (e: any) => void;
  onBlur?: (e: any) => void;
  onConfirm?: (e: any) => void;
}

export const Input: React.ComponentType<InputProps> = createHostComponent<InputProps>('input');

Input.defaultProps = {
  type: 'text',
  password: false,
  value: '',
  disabled: false,
  maxlength: 140,
  focus: false,
  selectionEnd: -1,
  selectionStart: -1,
  cursorSpacing: 0,
  cursor: -1,
};
