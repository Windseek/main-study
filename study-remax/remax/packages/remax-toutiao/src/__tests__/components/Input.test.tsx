import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { Input } from '../../hostComponents';

describe('Input', () => {
  it('render correctly', () => {
    const tree = TestRenderer.create(<Input value="" type="text" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
