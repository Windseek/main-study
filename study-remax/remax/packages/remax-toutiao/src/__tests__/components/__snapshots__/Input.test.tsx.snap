// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<input
  cursor={-1}
  cursorSpacing={0}
  disabled={false}
  focus={false}
  maxlength={140}
  password={false}
  selectionEnd={-1}
  selectionStart={-1}
  type="text"
  value=""
/>
`;
